app/models/concerns/host_searching.rb

```diff
@@ -109,13 +113,18 @@ def filter_by_compliance(_filter, operator, value)
     end
 
     def filter_by_compliance_score(_filter, operator, score)
-      ids = ::Host.includes(:test_result_profiles).select do |host|
-        host.compliance_score.public_send(operator, score.to_f)
+      unless NUM_OPERATORS.include?(operator)
+        raise ActiveRecord::StatementInvalid
       end
-      ids = ids.pluck(:id).map { |id| "'#{id}'" }
-      return { conditions: '1=0' } if ids.empty?
 
-      { conditions: "hosts.id IN(#{ids.join(',')})" }
+      profiles = Profile.current_scope.select(:id)
+
+      hosts = ::Host.joins(:latest_test_results)
```

Comment:
Does this join give expected results?
Here is the SQL:
```
SELECT "inventory"."hosts".* 
FROM "inventory"."hosts" 
INNER JOIN "test_results" ON "test_results"."host_id" = "inventory"."hosts"."id" 
WHERE "inventory"."hosts"."account" = '00023'
```

Reply:
There was a point in my tests when this was working as it should :see_no_evil: will investigate further...

Reply:
Okay, so `has_many` doesn't like `join` calls in their lambda :disappointed: as a solution we can approach this from the other end via `::TestResult.latest` and narrow it down via `profile_id` and score from there to get the `host_id`s